playwright/src/main/java/com/microsoft/playwright/impl/UrlMatcher.java

```diff
@@ -30,20 +32,14 @@
 import static com.microsoft.playwright.impl.Utils.toJsRegexFlags;
 
 class UrlMatcher {
-  final Object rawSource;
-  private final Predicate<String> predicate;
-
-  private static Predicate<String> toPredicate(Pattern pattern) {
-    return s -> pattern.matcher(s).find();
-  }
-
-  static UrlMatcher any() {
-    return new UrlMatcher((Object) null, null);
-  }
+  private final URL baseURL;
```

Comment:
Why not resolve the pattern in the constructor and avoid plumbing base url all the way through?

Reply:
We need this later on when I'll add this check:

https://github.com/microsoft/playwright/blob/edd789780ac473083f6fb0b8740faddc43b439ab/packages/playwright-core/src/utils/isomorphic/urlMatch.ts#L101-L103